src/commands/basemaps-github/make.cog.github.ts

```diff
@@ -164,28 +221,43 @@ export class MakeCogGithub {
+  async updateVectorTileSet(filename: string, layer: ConfigLayer, individual: boolean): Promise<void> {
     const gh = new GithubApi(this.repository);
     const branch = `feat/bot-config-vector-${this.imagery}`;
 
-    // Prepare new aerial tileset config
+    // Prepare new vector tileset config
     logger.info({ imagery: this.imagery }, 'GitHub: Get the master TileSet config file');
-    const tileSetPath = fsa.joinAll('config', 'tileset', `${filename}.json`);
-    const tileSetContent = await gh.getContent(tileSetPath);
-
-    // update the existing tileset
-    const existingTileSet = tileSetContent != null ? (JSON.parse(tileSetContent) as ConfigTileSetVector) : undefined;
-    const newTileSet = await this.prepareVectorTileSetConfig(layer, existingTileSet);
-
-    // skip pull request tileset prepare failure.
-    if (newTileSet == null) throw new Error('Failed to prepare new Vector tileSet.');
-
-    // Github
-    const title = `config(vector): Update the ${this.imagery} to ${filename} config file.`;
-    const content = await prettyPrint(JSON.stringify(newTileSet, null, 2), ConfigPrettierFormat);
-    const file = { path: tileSetPath, content };
-    // Github create pull request
-    await gh.createPullRequest(branch, title, botEmail, [file]);
+    if (individual) {
+      const tileSetPath = fsa.joinAll('config', 'tileset', `${filename}.json`);
+      const newTileSet = await this.prepareVectorTileSetConfig(layer, undefined);
+      // skip pull request tileset prepare failure.
+      if (newTileSet == null) throw new Error('Failed to prepare new Vector tileSet.');
+
+      // Github
+      const title = `config(vector): Update the ${this.imagery} to ${filename} config file.`;
+      const content = await prettyPrint(JSON.stringify(newTileSet, null, 2), ConfigPrettierFormat);
+      const file = { path: tileSetPath, content };
+      // Github create pull request
+      await gh.createPullRequest(branch, title, botEmail, [file]);
+    } else {
+      const tileSetPath = fsa.joinAll('config', 'tileset', `topographic.json`);
+      const tileSetContent = await gh.getContent(tileSetPath);
+      if (tileSetContent == null) throw new Error(`Failed to get config topographic from config repo.`);
+
+      // update the existing tileset
+      const existingTileSet = JSON.parse(tileSetContent) as ConfigTileSetVector;
+      const newTileSet = await this.prepareVectorTileSetConfig(layer, existingTileSet);
+
+      // skip pull request tileset prepare failure.
+      if (newTileSet == null) throw new Error('Failed to prepare new Vector tileSet.');
+
+      // Github
+      const title = `config(vector): Update the ${this.imagery} to ${filename} config file.`;
+      const content = await prettyPrint(JSON.stringify(newTileSet, null, 2), ConfigPrettierFormat);
+      const file = { path: tileSetPath, content };
+      // Github create pull request
+      await gh.createPullRequest(branch, title, botEmail, [file]);
```

Comment:
These lines seem to be identical in both branches, so should probably be outside the `if` block.

Reply:
This main differences here are the `[file]` which containers special `path` and `content` for the pull requests. So it will need be inside the if else block for each case. Otherwise, we might need make a `let` variable for the `file` which is not expected coding convention.

Reply:
Lines 260 through 267 are *identical* to 277 through 284, so they absolutely can be outside the `if` block.

Reply:
Yeah, but line 264 requires the `newTileSet` output from 259 and line 265 require `tileSetPath` from 258. Same as the else block. Unless we defined a `let tileSetPath;` and `let newTileSet;` outside the blocks forehead. But we should avoid let variables in typescript, then use const instead.

Another options, is just wrap these four line into a separated function. So we don't need see it repeated.

Reply:
I don't think we should avoid `let` if it means duplicating code.